shared/Query.d.ts

```diff
@@ -0,0 +1,19 @@
+export interface Pagination {
```

Comment:
Better use Partial while casting to this interface instead of making every prop optional